TypeCobol.Analysis/Graph/MultiBranchContext.cs

```diff
@@ -0,0 +1,248 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace TypeCobol.Analysis.Graph
+{
+    /// <summary>
+    /// The context of a Multi Branch instruction like IF or Evaluate.
+    /// </summary>
+    public class MultiBranchContext<N, D> : System.ICloneable
```

Comment:
This could be a problem then because we have to ensure all blocks or block indexes publicly exposed through this class are the 'final' ones otherwise we'll have to give access to the relocation map (which we don't want).
the `MultiBranchContext` may not be required with this new observer mechanism on the abstract interpretation.